.github/workflows/release-snapshot.yaml

```diff
@@ -0,0 +1,34 @@
+# This workflow publishes snapshot releases as an npm package. These releases aren't supposed to be reusable in `production` environments.
+# It's triggered manually in case a snapshot release would be needed for testing purposes.
+name: Release Snapshot
+
+on:
+  workflow_dispatch:
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  release-snapshot:
+    name: Release Snapshot
+    runs-on: blacksmith-2vcpu-ubuntu-2204
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      - name: Setup tools
+        uses: ./.github/actions/setup-tools
+        with:
+          install-nodejs: 'true'
+
+      - name: Build the modules
+        run: npm run build && npm run build:ts && npm run flatten && npm run clean:artifacts
+
+      - name: Set Snapshot Version
+        run: |
+          npm version 0.0.0-snapshot.$(git rev-parse --short HEAD) --git-tag-version=false
+
+      - name: Publish Snapshot to NPM
+        run: |
+          npm publish --no-git-checks --tag snapshot --access public
```

Comment:
Use npm release for the build and publish to avoid duplication

Reply:
create a new release-snapshot npm task to so the release cmds are defined in one place